libs/platform/src/lib/components/list/list.component.ts

```diff
@@ -0,0 +1,631 @@
+    ContentChildren, QueryList, HostBinding, ViewChild,
+    ElementRef, AfterContentInit, Output, EventEmitter,
+    HostListener, ChangeDetectorRef, OnInit, AfterViewInit, ContentChild, Self, Optional, SkipSelf, Host, OnDestroy
+} from '@angular/core';
+import { FocusKeyManager } from '@angular/cdk/a11y';
+import { UP_ARROW, DOWN_ARROW, ENTER, SPACE } from '@angular/cdk/keycodes';
+import { NgControl, NgForm } from '@angular/forms';
+import { SelectionModel } from '@angular/cdk/collections';
+import { Subscription, Subject, of } from 'rxjs';
+import { takeUntil, delay, tap } from 'rxjs/operators';
+import { BaseComponent } from '../base';
+import { BaseInput } from '../form/base.input';
+import { ListDataSource, isDataSource } from '../../domain/data-source';
+import { ContentDensity, FormFieldControl } from '../../components/form/form-control';
+import { BaseListItem, ListItemDef } from './base-list-item';
+import { ListConfig } from './list.config';
+import { FormField } from '../form/form-field';
+
+
+export type SelectionType = '' | 'multi' | 'single' | 'delete';
+export type ListType = 'inactive' | 'active' | 'detail';
+let nextListId = 0;
+let nextListGrpHeaderId = 0;
+export type FdpListDataSource<T> = ListDataSource<T> | T[];
+export class SelectionChangeEvent {
+    selectedItems: BaseListItem[];
+    index: number;
+}
+
+/**
+ * The List component represents a container for list item types.
+ * It is used to display a list features.
+ */
+@Component({
+    selector: 'fdp-list',
+    templateUrl: './list.component.html',
+    encapsulation: ViewEncapsulation.None,
+    styleUrls: ['./list.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [{ provide: FormFieldControl, useExisting: ListComponent, multi: true }]
+})
+
+
+export class ListComponent extends BaseInput implements OnInit, AfterViewInit, AfterContentInit, OnDestroy {
+
+    /**Enables lazy loadMore of data */
+    @Input()
+    loadMore: boolean;
+
+    /**Enables data load on scroll for true
+     * false: enables data loading on button
+     * click
+     */
+    @Input()
+    loadOnScroll: boolean;
+
+    /**Wait time for new items */
+    @Input()
+    delayTime: number;
+
+    /**Max height for viewport to display scroll */
+    @Input()
+    maxHeight: string;
+
+    /**Items to be loaded at once */
+    @Input()
+    itemSize: number;
+
+    /**Scroll offset percentage to trigger scroll event */
+    @Input()
+    scrollOffsetPercentage: number;
+
+    /** Whether list component has removed borders */
+    @Input()
+    noBorder = false;
+
+    @Input()
+    selection = false;
+
+    /** Whether list component has removed bottom borders */
+    @Input()
+    noSeperator: boolean;
+
+
+    /** The type of the selection. Types include:
+    *''| 'multi' | 'single'|'delete'.
+    * Leave empty for default ().'
+    * Default value is set to ''
+    */
+    @Input()
+    public selectionMode: SelectionType = '';
```

Comment:
Can you work with real values? I woudl not work with '' as default value. What is default ? I woudl try to name it. IS this single selection or noSelection at all ?

Reply:
updated to none